alvr/server_openvr/cpp/alvr_server/alvr_server.cpp

```diff
@@ -409,41 +339,50 @@ void SetTracking(
     FfiDeviceMotion headMotion,
     FfiHandData leftHandData,
     FfiHandData rightHandData,
-    const FfiBodyTracker* bodyTrackers,
+    const FfiDeviceMotion* bodyTrackers,
     int bodyTrackersCount
 ) {
     if (g_driver_provider.hmd) {
         g_driver_provider.hmd->OnPoseUpdated(targetTimestampNs, headMotion);
     }
 
     if (g_driver_provider.left_hand_tracker) {
-        g_driver_provider.left_hand_tracker->onPoseUpdate(
+        g_driver_provider.left_hand_tracker->OnPoseUpdate(
             targetTimestampNs, controllerPoseTimeOffsetS, leftHandData
         );
     }
 
     if (g_driver_provider.left_controller) {
-        g_driver_provider.left_controller->onPoseUpdate(
+        g_driver_provider.left_controller->OnPoseUpdate(
             targetTimestampNs, controllerPoseTimeOffsetS, leftHandData
         );
     }
 
     if (g_driver_provider.right_hand_tracker) {
-        g_driver_provider.right_hand_tracker->onPoseUpdate(
+        g_driver_provider.right_hand_tracker->OnPoseUpdate(
             targetTimestampNs, controllerPoseTimeOffsetS, rightHandData
         );
     }
 
     if (g_driver_provider.right_controller) {
-        g_driver_provider.right_controller->onPoseUpdate(
+        g_driver_provider.right_controller->OnPoseUpdate(
             targetTimestampNs, controllerPoseTimeOffsetS, rightHandData
         );
     }
 
     if (Settings::Instance().m_enableBodyTrackingFakeVive) {
-        for (int i = 0; i < bodyTrackersCount; i++) {
-            g_driver_provider.generic_trackers.at(bodyTrackers[i].trackerID)
-                ->OnPoseUpdated(targetTimestampNs, bodyTrackers[i]);
+        for (auto id : BODY_IDS) {
+            auto* maybeTracker = (FakeViveTracker*)g_driver_provider.tracked_devices.at(id);
+            if (maybeTracker) {
+                const FfiDeviceMotion* maybeMotion = nullptr;
+                for (int i = 0; i < bodyTrackersCount; i++) {
+                    if (bodyTrackers[i].deviceID == id) {
+                        maybeMotion = &bodyTrackers[i];
+                        break;
+                    }
+                }
```

Comment:
This logic should probably be inverted. It should go through all motions and attempt to get the tracked device from the motion id, or do we want to ensure we call `OnPoseUpdate` for all even if we don't have a motion?

Reply:
This was deliberate. Basically before we were always sending all trackers motions to c++ but mark some as not tracked. I inverted this by making so that when not tracked they are not passed to c++ at all. To make sure that unused trackers are disabled, we always need to call OnPoseUpdated, so we iterate frost on all tracker IDs.

Reply:
Still seems needlessly inefficient, but it shouldn't really matter. But in that case I actually prefer sending all, it seems a bit cleaner

Reply:
Well... cleaning it up was my reason too, with a different perspective. I thought that not sending the DeviceMotion at all is a better way of marking as not tracked. And this avoid the need for a separate TrackerMotion structure. the fact that OnPoseUpdated needs to be called always is a implementation detail. calling OnPoseUpdated with null as parameter to marks as not tracked seems a good pattern for me

Reply:
Also I think that chunk of code is really hard to read. What I'd prefer is that you collect them into a map first and then just access that instead of having the nested loops

Reply:
Fixed from discussion on Discord